jgiven-core/src/main/java/com/tngtech/jgiven/impl/ScenarioExecutor.java

```diff
@@ -260,6 +266,22 @@ StageState getStageState(Object stage) {
         return stageState;
     }
 
+    private String getStageNameOrGiven(Object stage, String lastExecutedClassName) throws NoSuchFieldException, IllegalAccessException {
+        String name = SingleStageNameFieldGetterSetter.getStageName(stage);
+
+        return name == null ? "GIVEN" : name;
```

Comment:
we need to test this against a scenario where 
```
class SomeTest {

  // copy Scenario methods here

  @ScenarioStage
  private Stage<T> notGiven;

  @Test
  public void test(){
    notGiven.I_run_a_nonconformant_JGiven_test();
  }
}

Reply:
a general problem I have with this setup is that we are using a magic keyword here  to make things work.  A casual reader does not know why, of all things, "GIVEN" is the default here.